build_posix/aclocal/strict.m4

```diff
@@ -41,38 +42,53 @@ AC_DEFUN([AM_GCC_WARNINGS], [
 	# GCC 4.7
 	#	WiredTiger uses anonymous structures/unions, a C11 extension,
 	#	turn off those warnings.
-	# GCC 6.X
-	#	Additional warning messages.
+	gcc5=0
+	gcc6=0
+	gcc7=0
+	gcc8=0
 	case "$1" in
 	[*4.7.[0-9]*])					# gcc4.7
-		w="$w -Wno-c11-extensions"
-		w="$w -Wunsafe-loop-optimizations";;
+		w="$w -Wno-c11-extensions";;
+	[*5.[0-9].[0-9]*])				# gcc5.X
+		gcc5=1;;
 	[*6.[0-9].[0-9]*])				# gcc6.X
-		w="$w -Wduplicated-cond"
+		gcc5=1
+		gcc6=1;;
+	[*7.[0-9].[0-9]*])				# gcc7.X
+		gcc5=1
+		gcc6=1
+		gcc7=1;;
+	[*8.[0-9].[0-9]*])				# gcc8.X
+		gcc5=1
+		gcc6=1
+		gcc7=1
+		gcc8=1;;
+	esac
+
+	if test $gcc5 -eq 1; then
 		w="$w -Wformat-signedness"
 		w="$w -Wjump-misses-init"
-		w="$w -Wlogical-op"
 		w="$w -Wredundant-decls"
-		w="$w -Wunsafe-loop-optimizations"
-		w="$w -Wunused-const-variable=2"
 		w="$w -Wunused-macros"
-		w="$w -Wvariadic-macros";;
-	[*7.[0-9].[0-9]*])				# gcc7.X
+		w="$w -Wvariadic-macros"
+	fi
+	if test $gcc6 -eq 1; then
+		w="$w -Wduplicated-cond"
+		w="$w -Wlogical-op"
+		w="$w -Wunused-const-variable=2"
+	fi
+	if test $gcc7 -eq 1; then
 		w="$w -Walloca"
 		w="$w -Walloc-zero"
 		w="$w -Wduplicated-branches"
-		w="$w -Wduplicated-cond"
 		w="$w -Wformat-overflow=2"
 		w="$w -Wformat-signedness"
```

Comment:
-Wformat-signedness flag is included by gcc5, can it be removed from gcc7's block?

Reply:
Thanks!